CONTRIBUTING.md

```diff
@@ -210,32 +355,134 @@ public boolean pushedByWater() {
+change between worlds. The latter is preferred whenever possible.
+
+### PaperConfig example
+
 ```java
 public static boolean saveEmptyScoreboardTeams = false;
 private static void saveEmptyScoreboardTeams() {
+    // This is called automatically!
+    // The name also doesn't matter.
     saveEmptyScoreboardTeams = getBoolean("settings.save-empty-scoreboard-teams", false);
 }
 ```
-Notice that the field is always public, but the setter is always private. This is important to the way the configuration generation system works. To access this value, reference it as you would any other static value:
+
+Notice that the field is always public, but the setter is always private. This
+is important to the way the configuration generation system works. To access
+this value, reference it as you would any other static value:
+
 ```java
 if (!PaperConfig.saveEmptyScoreboardTeams) {
 ```
 
-### PaperWorldConfig example:
+It is often preferred that you use the fully qualified name for the
+configuration class when accessing it, like so:
+`com.destroystokyo.paper.PaperConfig.valueHere`.  
+If this is preferred, a developer for Paper might just fix that before merging,
+but it's always nice if you make it a habit where you only need 1-2 lines
+changed.
+
+### PaperWorldConfig example
+
 ```java
 public boolean useInhabitedTime = true;
 private void useInhabitedTime() {
+    // This is called automatically!
+    // The name also doesn't matter.
     useInhabitedTime = getBoolean("use-chunk-inhabited-timer", true);
 }
 ```
-Again, notice that the field is always public, but the setter is always private. To access this value, you'll need an instance of the ```net.minecraft.World``` object:
+
+Again, notice that the field is always public, but the setter is always private.
+To access this value, you'll need an instance of the `net.minecraft.World`
+object:
 
 ```java
 return this.world.paperConfig.useInhabitedTime ? this.w : 0;
 ```
+
+## Frequently Asked Questions
+
+### I can't find the NMS file I need!
+
+By default, Paper (and upstream) only import files we make changes to. If you
+would like to make changes to a file that isn't present in `Paper-Server`'s
+source directory, you just need to add it to our import script ran during the
+patching process.
+
+1. Save (rebuild) any patches you are in the middle of working on! Their
+progress will be lost if you do not;
+1. Identify the name(s) of the file(s) you want to import.
+   - A complete list of all possible file names can be found at
+   `./work/Minecraft/$MCVER/spigot/net/minecraft/server`. You might find
+   [MiniMappingViewer] very useful to find the file you need.
+1. Open the file at `./scripts/importmcdev.sh` and add the name of your file to
+the script. Skip to the 2nd last header and follow the instructions there;
+1. Re-patch the server `./paper patch`;
+1. Edit away!
+
+❗ This change is temporary! **DO NOT COMMIT CHANGES TO THIS FILE!**  
```

Comment:
```suggestion
> ❗ This change is temporary! **DO NOT COMMIT CHANGES TO THIS FILE!**  
```
Placing noteworthy messages in blockquotes will make them more noticeable than just an emoji at the start of the sentence